File: Frontend/frontend-sgac/src/app/core/dto/evidencia-response-dto.ts
export interface EvidenciaResponseDTO {
  idEvidenciaRegistroActividad: number;
  nombreArchivo: string;
  rutaArchivo: string;

  /** mimeType permite inferir el tipo de archivo en UI */
  mimeType: string;

  tamanioBytes: number;

  /** En el backend puede venir null porque ya no es necesario en UI */
  tipoEvidencia?: string;

  idTipoEstadoEvidencia?: number;

  /** Se conserva por compatibilidad (antes se usaba este campo) */
  estadoEvidencia?: string;

  /** Nuevo: nombre del estado de evidencia (SUBIDO/REVISADO/APROBADO/RECHAZADO/OBSERVADO si aplica) */
  nombreEstadoEvidencia?: string;

  /** Fecha de subida (ISO yyyy-MM-dd) */
  fechaSubida: string;

  /** Observación/retroalimentación del docente sobre la evidencia */
  observacionDocente?: string;

  /** Fecha de observación (ISO yyyy-MM-dd) */
  fechaObservacion?: string;
}

File: Backend/backend-sgac/src/main/java/org/uteq/sgacfinal/dto/response/EvidenciaResponseDTO.java
package org.uteq.sgacfinal.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EvidenciaResponseDTO {
    private Integer idEvidenciaRegistroActividad;
    private String nombreArchivo;
    private String rutaArchivo;
    private String mimeType;
    private Integer tamanioBytes;
    private LocalDate fechaSubida;

    private Integer idTipoEvidencia;
    private String tipoEvidencia;

    private Integer idTipoEstadoEvidencia;
    private String estadoEvidencia;

    // Nuevo: nombre del estado de evidencia (tabla ayudantia.tipo_estado_evidencia)
    private String nombreEstadoEvidencia;

    // Nuevo: observación/retroalimentación del docente sobre la evidencia
    private String observacionDocente;

    // Nuevo: fecha/hora de la observación
    private LocalDate fechaObservacion;
}
